packages/cli/src/controllers/invitation.controller.ts

```diff
@@ -0,0 +1,301 @@
+							email,
+							globalRole: role,
+						});
+						const savedUser = await transactionManager.save<User>(newUser);
+						createUsers[savedUser.email] = savedUser.id;
+						return savedUser;
+					}),
+				),
+			);
+		} catch (error) {
+			ErrorReporter.error(error);
+			this.logger.error('Failed to create user shells', { userShells: createUsers });
+			throw new InternalServerError('An error occurred during user creation');
+		}
+
+		this.logger.debug('Created user shell(s) successfully', { userId: req.user.id });
+		this.logger.verbose(total > 1 ? `${total} user shells created` : '1 user shell created', {
+			userShells: createUsers,
+		});
+
+		const baseUrl = getInstanceBaseUrl();
+
+		const usersPendingSetup = Object.entries(createUsers).filter(([email, id]) => id && email);
+
+		// send invite email to new or not yet setup users
+
+		const emailingResults = await Promise.all(
+			usersPendingSetup.map(async ([email, id]) => {
+				if (!id) {
+					// This should never happen since those are removed from the list before reaching this point
+					throw new InternalServerError('User ID is missing for user with email address');
+				}
+				const inviteAcceptUrl = generateUserInviteUrl(req.user.id, id);
+				const resp: {
+					user: { id: string | null; email: string; inviteAcceptUrl?: string; emailSent: boolean };
+					error?: string;
+				} = {
+					user: {
+						id,
+						email,
+						inviteAcceptUrl,
+						emailSent: false,
+					},
+				};
+				try {
+					const result = await this.mailer.invite({
+						email,
+						inviteAcceptUrl,
+						domain: baseUrl,
+					});
+					if (result.emailSent) {
+						resp.user.emailSent = true;
+						delete resp.user.inviteAcceptUrl;
+						void this.internalHooks.onUserTransactionalEmail({
+							user_id: id,
+							message_type: 'New user invite',
+							public_api: false,
+						});
+					}
+
+					void this.internalHooks.onUserInvite({
+						user: req.user,
+						target_user_id: Object.values(createUsers) as string[],
+						public_api: false,
+						email_sent: result.emailSent,
+					});
+				} catch (error) {
+					if (error instanceof Error) {
+						void this.internalHooks.onEmailFailed({
+							user: req.user,
+							message_type: 'New user invite',
+							public_api: false,
+						});
+						this.logger.error('Failed to send email', {
+							userId: req.user.id,
+							inviteAcceptUrl,
+							domain: baseUrl,
+							email,
+						});
+						resp.error = error.message;
+					}
+				}
+				return resp;
+			}),
+		);
+
+		await this.externalHooks.run('user.invited', [usersToSetUp]);
+
+		this.logger.debug(
+			usersPendingSetup.length > 1
+				? `Sent ${usersPendingSetup.length} invite emails successfully`
+				: 'Sent 1 invite email successfully',
+			{ userShells: createUsers },
+		);
+
+		return emailingResults;
+	}
+
+	/**
+	 * Fill out user shell with first name, last name, and password.
+	 */
+	@NoAuthRequired()
```

Comment:
For some reason this decorator is not working and the route is always authenticated. @netroy any idea what might be happening?

Reply:
This is bug in the legacy auth middleware, that prevents requests from reaching these routers, and since that middleware is applied globally, we can't exclude routes because `req.route` does not exist in that middleware.
This is why we've had [this check](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/middlewares/auth.ts#L92) in place.
I suggest that we update [these lines](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/middlewares/auth.ts#L65-L66) to match the new URL.
Once we have all the old route out of the main express app, and into dedicated routers, we can then get rid of the global auth middleware, and all the checks like these.